source/docs/software/kinematics-and-odometry/differential-drive-odometry.rst

```diff
@@ -20,55 +23,60 @@ The ``DifferentialDriveOdometry`` class requires one mandatory argument and one
       // our starting pose is 5 meters along the long end of the field and in the
       // center of the field along the short end, facing forward.
       DifferentialDriveOdometry m_odometry = new DifferentialDriveOdometry(
-        getGyroHeading(), new Pose2d(5.0, 13.5, new Rotation2d()));
+        m_gyro.getRotation2d(),
+        m_leftEncoder.getDistance(), m_rightEncoder.getDistance(),
+        new Pose2d(5.0, 13.5, new Rotation2d()));
 
    .. code-tab:: c++
 
       // Creating my odometry object. Here,
       // our starting pose is 5 meters along the long end of the field and in the
       // center of the field along the short end, facing forward.
-      frc::DifferentialDriveOdometry m_odometry{GetGyroHeading(),
+      frc::DifferentialDriveOdometry m_odometry{
+        m_gyro.GetRotation2d(),
+        units::meter_t{m_leftEncoder.GetDistance()},
+        units::meter_t{m_rightEncoder.GetDistance()},
         frc::Pose2d{5_m, 13.5_m, 0_rad}};
 
 
 Updating the Robot Pose
 -----------------------
 The ``update`` method can be used to update the robot's position on the field. This method must be called periodically, preferably in the ``periodic()`` method of a :ref:`Subsystem <docs/software/commandbased/subsystems:Subsystems>`. The ``update`` method returns the new updated pose of the robot. This method takes in the gyro angle of the robot, along with the left encoder distance and right encoder distance.
 
-.. note:: The encoder distances in Java must be in meters. In C++, the units library can be used to represent the distance using any linear distance unit. If the robot is moving forward in a straight line, **both** distances (left and right) must be positive.
+.. note:: If the robot is moving forward in a straight line, **both** distances (left and right) must be increasing.
 
 .. tabs::
 
    .. code-tab:: java
 
       @Override
       public void periodic() {
-        // Get my gyro angle. We are negating the value because gyros return positive
-        // values as the robot turns clockwise. This is not standard convention that is
-        // used by the WPILib classes.
-        var gyroAngle = Rotation2d.fromDegrees(-m_gyro.getAngle());
+        // Get my gyro angle.
+        var gyroAngle = m_gyro.getRotation2d();
 
         // Update the pose
-        m_pose = m_odometry.update(gyroAngle, m_leftEncoder.getDistance(), m_rightEncoder.getDistance());
+        m_pose = m_odometry.update(gyroAngle,
+          m_leftEncoder.getDistance(),
+          m_rightEncoder.getDistance());
       }
 
    .. code-tab:: c++
 
       void Periodic() override {
-        // Get my gyro angle. We are negating the value because gyros return positive
-        // values as the robot turns clockwise. This is not standard convention that is
-        // used by the WPILib classes.
-        frc::Rotation2d gyroAngle{units::degree_t(-m_gyro.GetAngle())};
+        // Get my gyro angle.
+        frc::Rotation2d gyroAngle = m_gyro.GetRotation2d();
```

Comment:
Modify comment. We don't want users to assuming rotation2D *is* a gyro angle, although they mostly can be used the same.

Reply:
The rotation2d represents the gyro angle, and any place that an angle is used wpilib takes a rotation2d- what's the issue?

Reply:
PID on gyro angle without Rotation2d is a common thing, so I disagree with that statement. But to clarify, a Rotation2d represents a rotation and a rotation does not have to be provided by a gyro, even if it usually is. I'd refactor the comment to "retrieve the rotation of the robot from the gyro" or something similar.

Reply:
Sure, any rotation doesn't have to be provided by a gyro, but in this case it is- the variable is storing the gyro angle. I don't see how naming the variable after the type instead of represents increases clarity.